elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/verify/DynamicConfigVerifier.java

```diff
@@ -0,0 +1,159 @@
+        if (!cli.hasOption("tarFile") || !cli.hasOption("signatureFile") || !cli.hasOption("publicKeyName")) {
+            printHelp(options);
+            throw new MissingOptionException("Missing required option");
+        }
+
+        String modelTarFile = cli.getOptionValue("tarFile");
+        String signatureFile = cli.getOptionValue("signatureFile");
+        String publicKeyName = cli.getOptionValue("publicKeyName");
+        long tarFileSize = getFileSize(modelTarFile);
+
+        if (verify(readTarContents(modelTarFile), tarFileSize,
+                signatureFile, getPublicKey(publicKeyName))) {
+            log.info("Successfully Validated " + modelTarFile);
+        }
+        else {
+            log.error("Could not verify " + modelTarFile + " with details provided");
+        }
+    }
+
+    /**
+     * Verify signature of tar.gz.
+     * @param fileList : list of files
+     * @param sizeOfFile : size of tar file
+     * @param signature : file containing signature
+     * @param publicKey : public key name
+     * @return whether the file can be verified by given key and signature
+     * @throws NoSuchAlgorithmException
+     * @throws InvalidKeyException
+     * @throws SignatureException
+     */
+    public static boolean verify(String fileList, long sizeOfFile, String signature, PublicKey publicKey)
+            throws NoSuchAlgorithmException, InvalidKeyException, SignatureException {
+
+        Signature publicSignature;
+
+        publicSignature = Signature.getInstance("SHA256withRSA");
+        publicSignature.initVerify(publicKey);
+        publicSignature.update((fileList + sizeOfFile).getBytes(StandardCharsets.UTF_8));
+        byte[] signatureBytes = Base64.getDecoder().decode(signature);
+        return publicSignature.verify(signatureBytes);
+    }
+
+    /**
+     * Get tar file size.
+     * @param modelTarFile
+     * @return size of tar file
+     */
+    private static long getFileSize(String modelTarFile) {
+        return FileUtils.sizeOf(new File(modelTarFile));
+    }
+
+    /**
+     * Retrieve public key from Key Store.
+     * @param keyName : name of the public key
+     * @return publickey
+     */
+    private static PublicKey getPublicKey(String keyName) throws KeyStoreException {
+        PublicKey publicKey = null;
+        KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        Certificate cert = keyStore.getCertificate(keyName);
+        publicKey = cert.getPublicKey();
+        return publicKey;
+    }
+
+    private static String readTarContents(String archiveFile) throws FileNotFoundException, IOException {
+        StringBuffer sb = new StringBuffer();
+        TarArchiveInputStream archive = new TarArchiveInputStream(
+                new GzipCompressorInputStream(new BufferedInputStream(new FileInputStream(archiveFile))));
+        TarArchiveEntry entry;
+        while ((entry = archive.getNextTarEntry()) != null) {
+            sb.append(entry.getName());
```

Comment:
This seems to be only verifying the names of the files but not the file contents.  Am I reading this correct?   If so - we'll need to verify the contents as well.